crates/bevy_ecs/src/bundle.rs

```diff
@@ -1346,11 +1349,39 @@ impl Bundles {
                 // - it was created in the same order as the components in T
                 unsafe { BundleInfo::new(core::any::type_name::<T>(), components, component_ids, id) };
             bundle_infos.push(bundle_info);
+
+            // Be sure to update component bundle with requires
+            contributed_bundle_ids.remove(&TypeId::of::<T>());
+
```

Comment:
Is this really necessary? I don't see how `contributed_bundle_ids` could exist for `T` if it hasn't already been registered. Unless I'm missing something, this feels like "defensive coding" protecting against incorrect internal code. Remove ops are expensive enough that we should remove this if it isn't functionally required. 

```suggestion
```

Reply:
It's defense programming. Really it's a slowdown for no apparent reason